### PR TITLE
[BEAM-4530]  Remove 'Executor' concept

### PR DESCRIPTION
**Pull request by VaclavPlajt**
Executors concept removed.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand:
   - [x] What the pull request does
   - [x] Why it does it
   - [x] How it does it
   - [x] Why this approach
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Run `./gradlew build` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by mareksimunek** [comment]
LGTM

File: sdks/java/extensions/euphoria/euphoria-fluent/src/main/java/org/apache/beam/sdk/extensions/euphoria/fluent/Dataset.java [hunk 1]
```diff
@@ -70,7 +69,7 @@
     return this;
   }
 
-  public void execute(Executor exec) throws Exception {
-    exec.submit(this.wrap.getFlow()).get();
-  }
+//  public void execute(BeamRunnerWrapper runner) throws Exception {
```
Comment (dmvk) [inline sdks/java/extensions/euphoria/euphoria-fluent/src/main/java/org/apache/beam/sdk/extensions/euphoria/fluent/Dataset.java]: can we delete this?

File: sdks/java/extensions/euphoria/euphoria-beam/src/main/java/org/apache/beam/sdk/extensions/euphoria/beam/BeamRunnerWrapper.java [hunk 2]
```diff
@@ -17,41 +17,62 @@
...
+  public static BeamRunnerWrapper ofDirect() {
+    final String[] args = {"--runner=DirectRunner"};
+    final PipelineOptions options = PipelineOptionsFactory.fromArgs(args).as(PipelineOptions.class);
+    return new BeamRunnerWrapper(options).withAllowedLateness(java.time.Duration.ofHours(1));
+  }
+
+  /**
+   * Blocks until a given {@link Flow} is executed.
+   */
+  public Result executeSync(Flow flow) {
```
Comment (dmvk) [inline sdks/java/extensions/euphoria/euphoria-beam/src/main/java/org/apache/beam/sdk/extensions/euphoria/beam/BeamRunnerWrapper.java:75]: we should have consistent naming here `executeSync` vs `submitAsync`

**Review by dmvk — CHANGES_REQUESTED** [review]
LGTM 👍 I only have one minor comment.

File: sdks/java/extensions/euphoria/euphoria-beam/src/main/java/org/apache/beam/sdk/extensions/euphoria/beam/BeamRunnerWrapper.java (hunk 2 above)
Comment (VaclavPlajt) [reply sdks/java/extensions/euphoria/euphoria-beam/src/main/java/org/apache/beam/sdk/extensions/euphoria/beam/BeamRunnerWrapper.java:75]: You are right. Renamed to `executeSync`, `executeAsync`.

File: sdks/java/extensions/euphoria/euphoria-fluent/src/main/java/org/apache/beam/sdk/extensions/euphoria/fluent/Dataset.java (hunk 1 above)
Comment (VaclavPlajt) [reply sdks/java/extensions/euphoria/euphoria-fluent/src/main/java/org/apache/beam/sdk/extensions/euphoria/fluent/Dataset.java]: Deleted.

**Review by dmvk — APPROVED** [review]

**Pull request merged**